### ISSUE TITLE
RangeSlider setSpan can't cross old range's boundaries

### ISSUE THREAD

**Issue by jamshark70** [supercollider/supercollider#3616]
```
(
{
	var range = RangeSlider(nil, Rect(800, 200, 200, 30)).setSpan(0.25, 0.25).front;
	0.5.wait;
	range.setSpan(0.75, 0.75);
}.fork(AppClock);
)
```

Final displayed range is 0.25 .. 0.75. :roll_eyes: 

Workaround: Do it twice. :open_mouth:

**Comment by patrickdupuis** [comment]
Could it be that `hi` needs to be moved first?
```supercollider
	setSpan { arg lo, hi;
		this.lo_(lo);
		this.hi_(hi);
	}
```

**Comment by patrickdupuis** [comment]
Swapping `lo` and `hi` here seems to fix it: https://github.com/supercollider/supercollider/blob/8c3563a8065cb623087f267dfe50e228224a4572/SCClassLibrary/Common/GUI/Base/QRangeSlider.sc#L68-L71

**Comment by patrickdupuis** [comment]
We will need some logic here to determine which boundary should moves first.

**Comment by patrickdupuis** [comment]
`setSpan` already seems to be able to handle cases where `hi < lo`, which is a good thing. I think all that is needed is something like:
```supercollider
    setSpan { |lo, hi|
        var oldHigh = this.hi;

        if (lo <= oldHigh) {
            this.lo_(lo);
            this.hi_(hi);
        } {
            this.hi_(hi);
            this.lo_(lo);
        }
    }
```

**Comment by patrickdupuis** [comment]
nope, the above still needs work.

**Comment by patrickdupuis** [comment]
Fixed in #3620

**Comment by brianlheim** [comment]
@patrickdupuis - typically we only close after a fixing PR has been merged
